### PR TITLE
[4.3.0] Update about-mediators.md

### PR DESCRIPTION
**Pull request by DinithiDiaz**
## Purpose
This PR is to update the list of mediators. I have updated the mediator categorization according to the mediator palette in the MI for VS Code extension.

This will fix https://github.com/wso2/docs-mi/issues/837.

### CODE REVIEW

File: en/docs/reference/mediators/about-mediators.md [hunk 1]
```diff
@@ -38,26 +38,70 @@ Mediators are classified as follows based on whether or not they access the mess
...
+- [Fault mediator]({{base_path}}/reference/mediators/fault-mediator)
+- [PayloadFactory mediator]({{base_path}}/reference/mediators/payloadfactory-mediator)
+- [Data Mapper mediator]({{base_path}}/reference/mediators/data-mapper-mediator)
+
+### Extension mediators
+
+- [Script mediator]({{base_path}}/reference/mediators/script-mediator)
+- [Class mediator]({{base_path}}/reference/mediators/class-mediator)
+
+### Generic mediators
```
Comment (chanikag) [inline en/docs/reference/mediators/about-mediators.md:70]: Shall we move this to top, as these are widely used.

Comment (DinithiDiaz) [reply en/docs/reference/mediators/about-mediators.md]: Fixed via https://github.com/wso2/docs-mi/pull/852/commits/386fc20e8778245e9abeb8dccd84e68ba2b67a6a.

File: en/docs/reference/mediators/about-mediators.md [hunk 2]
```diff
@@ -38,26 +38,70 @@ Mediators are classified as follows based on whether or not they access the mess
...
-
-**Extension Mediators**
-
-[Class]({{base_path}}/reference/mediators/class-mediator) | [Script]({{base_path}}/reference/mediators/script-mediator) |
+## List of mediators
+
+WSO2 Micro Integrator includes a comprehensive library of mediators that provide functionality for implementing widely used [Enterprise Integration Patterns (EIPs)]({{base_path}}/learn/enterprise-integration-patterns/eip-overview/). You can also easily write a custom mediator to provide additional functionality using various technologies such as Java, scripting, and Spring.
```
Comment (chanikag) [inline en/docs/reference/mediators/about-mediators.md:43]: Let's not mention about Spring. We have deprecated this mediator (Spring Mediator) and all the relevant docs are removed.

Comment (DinithiDiaz) [inline en/docs/reference/mediators/about-mediators.md:43]:
```suggestion
WSO2 Micro Integrator includes a comprehensive library of mediators that provide functionality for implementing widely used [Enterprise Integration Patterns (EIPs)]({{base_path}}/learn/enterprise-integration-patterns/eip-overview/). You can also easily write a custom mediator to provide additional functionality using various technologies such as Java and scripting.
```

Comment (DinithiDiaz) [reply en/docs/reference/mediators/about-mediators.md]: Removed via https://github.com/wso2/docs-mi/pull/852/commits/a5ee30bc2f229b92138d9359522181bf80c788dd.

**Review by chanikag — APPROVED** [review]

**Pull request merged**